middleware.go

```diff
@@ -32,41 +32,60 @@ func (r *Router) useInterface(mw middleware) {
 	r.middlewares = append(r.middlewares, mw)
 }
 
-// CORSMethodMiddleware sets the Access-Control-Allow-Methods response header
-// on a request, by matching routes based only on paths. It also handles
-// OPTIONS requests, by settings Access-Control-Allow-Methods, and then
-// returning without calling the next http handler.
+// CORSMethodMiddleware automatically sets the Access-Control-Allow-Methods response header
+// on preflight requests for routes that have an OPTIONS method matcher to all the method
```

Comment:
I just realized this behavior is wrong. CORS headers should be set on all requests, including non-preflight.